### ISSUE TITLE
Docker image consume high memory when even when empty

### ISSUE THREAD

**Issue by tristone13th** [typesense/typesense#930]
## Description

I run the docker image without any records put into it, and it consumes about 100MiB memory.

![image](https://user-images.githubusercontent.com/17382962/224474287-a7e15ae7-292d-42e2-a807-7a9f913574a0.png)

## Steps to reproduce

The docker file:

```dockerfile
FROM typesense/typesense:0.23.0
EXPOSE 8108
WORKDIR /opt
RUN mkdir /data
CMD [ "typesense-server", "--data-dir=/data", "--api-key=xxx", "--enable-cors", "--api-port=8108" ]
```

## Metadata

**Typesense Version**: 0.23.0

**Comment by kishorenc** [comment]
There is a baseline memory usage of about 35 MB (mostly buffers/caches that are pre-allocated). So, I don't know where that 100 MiB number is coming from. Can you try running the binary directly and measure usage via htop?

**Comment by tristone13th** [comment]
@kishorenc This is from the bare binary.

![image](https://user-images.githubusercontent.com/17382962/224521803-ea4fba86-aca3-4565-a353-6c357b601a1e.png)

**Comment by kishorenc** [comment]
Yes, in both cases, it shows an usage of about 45 MB of RAM -- that's inline with the expectation. I'm not sure why Docker will show an higher value.
